### PR TITLE
Publish Helm charts as OCI artifacts

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area delivery
/kind enhancement

**What this PR does / why we need it**:
We should start publishing Helm charts as OCI artifacts that we can deploy them as `Extension` in the future (see https://github.com/gardener/gardener/issues/9635).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature operator
Helm charts of extension and admission controller are published as OCI artifacts now.
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
This PR proposes changes that would break the pipeline definition:
```
<invalid YAML>: found undefined alias 'registry-cache'
  in "<unicode string>", line 66, column 13:
              - *registry-cache
                ^
```

**Comment by gardener-robot-ci-3** [comment]
The pipeline-definition has been fixed.

**Comment by oliver-goetz** [comment]
The PR is now similar to the solution for the `registry-cache` https://github.com/gardener/gardener-extension-registry-cache/pull/229 so that all images are built in concourse now.

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -17,6 +61,21 @@ gardener-extension-shoot-rsyslog-relp:
...
+    pull-request:
+      traits:
+        pull-request: ~
+        options:
+          public_build_logs: true
+        publish:
+          helmcharts:
+          - *shoot-rsyslog-relp
+          - *shoot-rsyslog-relp-admission-application
+          - *shoot-rsyslog-relp-admission-runtime
```
Comment (ialidzhikov) [inline .ci/pipeline_definitions:78]:
```suggestion
```

See https://github.com/gardener/gardener-extension-registry-cache/pull/258

**Comment by plkokanov** [comment]
Checking the kubelet logs from the https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener-extension-shoot-rsyslog-relp/147/pull-gardener-extension-shoot-rsyslog-relp-e2e-kind/1836851466954870784 test failure it seems that the reason is:
```
Sep 19 19:54:46 gardener-local-control-plane kubelet[804]: E0919 19:54:46.133268     804 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932\": failed to resolve reference \"docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932\": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed" image="local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932"
Sep 19 19:54:46 gardener-local-control-plane kubelet[804]: E0919 19:54:46.133322     804 kuberuntime_image.go:55] "Failed to pull image" err="failed to pull and unpack image \"docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932\": failed to resolve reference \"docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932\": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed" image="local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-35-g2cd3932"
```

**Comment by plkokanov** [comment]
For some reason the test above fails because skaffold uploads images in kind with the following tags:
```
Tags used in deployment:
 - local-skaffold/gardener-extension-shoot-rsyslog-relp-echo-server -> local-skaffold/gardener-extension-shoot-rsyslog-relp-echo-server:a55ecfef8dc2602fd9a0d097996de1e5946b7e57b0447f9ed73055080160e860
 - local-skaffold/gardener-extension-shoot-rsyslog-relp -> local-skaffold/gardener-extension-shoot-rsyslog-relp:e1e2c9d72c4d07003d8eca0dfafd5d66c674ca52b21af644a84a7e0e9d6206f1
 - local-skaffold/gardener-extension-shoot-rsyslog-relp-admission -> local-skaffold/gardener-extension-shoot-rsyslog-relp-admission:7915d63f10a6e9f6900ea5f54786c4251eb39f620f0ee8d3de6c90a87
```

However, in the `ControllerDeployment` the image for `gardener-extension-shoot-rsyslog-relp` is `local-skaffold/gardener-extension-shoot-rsyslog-relp:v0.5.0-37-g635821f`

Note that skaffold does mention that it adds the `v0.5.0-37-g635821f` tags:
```
Loading index.docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:a92ac8a6aa48e9be0aaec1e0e2ea4b73f521d82ceece6855eceaa7bde0adc633
Loaded index.docker.io/local-skaffold/gardener-extension-shoot-rsyslog-relp:a92ac8a6aa48e9be0aaec1e0e2ea4b73f521d82ceece6855eceaa7bde0adc633
Adding tag v0.5.0-37-g635821f
Added tag v0.5.0-37-g635821f
Starting post-build hooks...
Completed post-build hooks
Build [local-skaffold/gardener-extension-shoot-rsyslog-relp] succeeded
```

This also happens when trying to setup the extension locally.

The reason seems to be that we do not explicitly specify `SKAFFOLD_DEFAULT_REPO = localhost:5001` and  `SKAFFOLD_PUSH = true` in the Makefile

@oliver-goetz could you please add the following changes to the Makefile in this PR:

```patch
diff --git a/Makefile b/Makefile
index 8a1b65c..eee1e20 100644
--- a/Makefile
+++ b/Makefile
@@ -140,8 +140,12 @@ test-e2e-local: $(GINKGO)
 ci-e2e-kind: $(KIND) $(YQ)
        ./hack/ci-e2e-kind.sh
 
+# # speed-up skaffold deployments by building all images concurrently
+export SKAFFOLD_BUILD_CONCURRENCY = 0
+extension-up extension-dev: export SKAFFOLD_DEFAULT_REPO = localhost:5001
+extension-up extension-dev: export SKAFFOLD_PUSH = true
 # use static label for skaffold to prevent rolling all gardener components on every `skaffold` invocation
-extension-up extension-down: export SKAFFOLD_LABEL = skaffold.dev/run-id=extension-local
+extension-up extension-dev extension-down: export SKAFFOLD_LABEL = skaffold.dev/run-id=extension-local
 
 extension-up: $(SKAFFOLD) $(HELM) $(KUBECTL) $(KIND)
        @LD_FLAGS=$(LD_FLAGS) $(SKAFFOLD) run
```

**Comment by oliver-goetz** [comment]
Done 👍 
Thanks for investigating the issue 🙂

**Pull request merged**